### PR TITLE
Add tests for functions in utilities/data.py

### PR DESCRIPTION
**Pull request by roshikouhai**
## What does this PR do?

Add unit tests for functions `has_iterable_dataset`, `has_len` and `get_len` in utilities/data.py. 

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/utilities/test_data.py [hunk 1]
```diff
@@ -19,3 +22,51 @@ def test_extract_batch_size():
...
+    class MockDatasetWithLen(Dataset):
+        def __len__(self):
+            return 1
+
+    assert get_len(DataLoader(MockDatasetWithLen())) == 1
+
+    value = get_len(DataLoader(IterableDataset()))
+
+    assert isinstance(value, float)
+    assert value == float("inf")
```
Comment (ananthsub) [inline tests/utilities/test_data.py:72]: you can use these datasets from the test helpers for dataset vs iterable datasets: https://github.com/PyTorchLightning/pytorch-lightning/blob/e54180363663b690367f0051484650fcd12e5de2/tests/helpers/boring_model.py#L36-L69

the deepsource linter is also warning that `Method '__getitem__' is abstract in class 'Dataset' but is not overridden`: https://deepsource.io/gh/PyTorchLightning/pytorch-lightning/run/5241bb94-206a-464b-8a2b-b2d14afa08c0/python/PYL-W0223 

using the above utiities should fix them. if there are missing variants, you can add to the test helpers too

**Review by awaelchli — APPROVED** [review]
LGTM, agree would be great to incorporate @ananthsub 's suggestion

File: tests/utilities/test_data.py [hunk 2]
```diff
@@ -19,3 +22,51 @@ def test_extract_batch_size():
...
+        def __len__(self):
+            return 1
+
+    assert has_len(DataLoader(MockDatasetWithLen()))
+
+    class MockDatasetWithZeroLen(Dataset):
+        def __len__(self):
+            return 0
+
+    with pytest.raises(ValueError):
```
Comment (awaelchli) [inline tests/utilities/test_data.py:56]:
```suggestion
    with pytest.raises(ValueError, match="`Dataloader` returned 0 length."):
```

**Review by ananthsub — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
